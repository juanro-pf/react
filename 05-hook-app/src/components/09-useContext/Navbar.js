import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink activeClassName='active' exact to="/" className="nav-link" >Home</NavLink>
                    <NavLink activeClassName='active' exact to="/login" className="nav-link" >Login</NavLink>
                    <NavLink activeClassName='active' exact to="/about" className="nav-link" >About</NavLink>
                </div>
            </div>
        </nav>
    );
};
