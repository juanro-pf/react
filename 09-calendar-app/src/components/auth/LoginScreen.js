import React from 'react';
import './login.css';
import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../../actions/auth';
import Swal from 'sweetalert2';

export const LoginScreen = () => {

  const dispatch = useDispatch();

  const [ formLoginValues, handleLoginInputChange ]= useForm({
    lEmail: 'jro@gmail.com',
    lPassword: '123456'
  });

  const [ formRegisterValues, handleRegisterInputChange ]= useForm({
    rName: 'Ropo',
    rEmail: 'jropo@gmail.com',
    rPassword: '123456',
    rPassword2: '123456'
  });

  const { lEmail, lPassword }= formLoginValues;
  const { rEmail, rName , rPassword , rPassword2 }= formRegisterValues;

  const handleLogin= (e) => {
    e.preventDefault();
    dispatch( startLogin( lEmail, lPassword ) );
  };

  const handleRegister= (e) => {
    e.preventDefault();

    if( rPassword !== rPassword2 ){
      return Swal.fire('Error', 'Passwords must match', 'error');
    }
    
    dispatch( startRegister( rEmail, rName, rPassword ) );
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Log in</h3>
          <form onSubmit= { handleLogin }>
            <div className="form-group">
              <input 
                type="text"
                className="form-control"
                placeholder="Email"
                name= 'lEmail'
                value= { lEmail }
                onChange= { handleLoginInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                name= 'lPassword'
                value= { lPassword }
                onChange= { handleLoginInputChange }
              />
            </div>
            <div className="form-group">
              <input 
                type="submit"
                className="btnSubmit"
                value="Login" 
              />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Register</h3>
          <form onSubmit= { handleRegister }>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Name"
                name= 'rName'
                value= { rName }
                onChange= { handleRegisterInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Email"
                name= 'rEmail'
                value= { rEmail }
                onChange= { handleRegisterInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password" 
                name= 'rPassword'
                value= { rPassword }
                onChange= { handleRegisterInputChange }
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repeat password"
                name= 'rPassword2'
                value= { rPassword2 }
                onChange= { handleRegisterInputChange }
              />
            </div>

            <div className="form-group">
              <input 
                type="submit" 
                className="btnSubmit" 
                value="Create account"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  )
}