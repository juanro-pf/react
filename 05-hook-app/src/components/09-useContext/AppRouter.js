import React from 'react';
import { Navbar } from './Navbar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <div>

                    <Navbar />

                    <div className='container'>
                        <Switch>
                            <Route exact path='/' component= { HomeScreen } />
                            <Route exact path='/login' component= { LoginScreen } />
                            <Route exact path='/about' component= { AboutScreen } />
                            {/* <Route component= { HomeScreen } /> */}
                            <Redirect to='./' />
                        </Switch>
                    </div>

                </div>
            </Router>
        </div>
    );
};
