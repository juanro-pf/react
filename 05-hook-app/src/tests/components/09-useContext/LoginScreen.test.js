import React from 'react';
import { act } from '@testing-library/react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

import '@testing-library/jest-dom';

describe('Testing LoginScreen', () => {

    const setUser= jest.fn();
    
    //mount se usa para ver tambien los componentes dentro de componentes (HomeScreen dentro de Usecontext.Provider)
    const wrapper= mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );
    
    test('Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('setUser with desired argument', () => {
        
        wrapper.find('button').simulate('click');

        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toHaveBeenCalledWith({
            id: 1234,
            name: 'Juan Ro'
        });

    });
    

});
