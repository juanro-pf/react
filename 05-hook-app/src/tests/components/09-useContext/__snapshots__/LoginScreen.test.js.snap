// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen Snapshot 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
